Automattic/woocommerce-payments
# Fix incorrect amounts on payment request button for zero decimal currencies

Fixes #3376


#### Changes proposed in this Pull Request
If the currency is set to Japanese yen (or any other currency listed [here](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments-utils.php#L193)) the amounts shown are multiplied by 100 on Payment Request sheet in product page, cart page, or checkout page.  This is because the currency is not being passed into the `prepare_amount` method, which makes it use the incorrect conversion rate that is used for non-zero decimal currencies. This change passes in the value of `get_woocommerce_currency` to the above method








#### Testing instructions





* Add Japanese yen to the list of currencies in WooCommerce -> Settings -> Multi-currency
* On the frontend, under My Account > Account Details, switch your currency to JPY and save. (Or use currency switcher widget)
* Go to a product page
* Click the payment request button
* Ensure that the prices shown matches the correct amount for the product


Before
<img width="827" alt="Screen Shot 2021-11-15 at 2 32 07 PM" src="https://user-images.githubusercontent.com/6216000/141753170-edb20b93-1072-4ee5-a713-270e11920cb8.png">

After
<img width="827" alt="Screen Shot 2021-11-15 at 2 30 03 PM" src="https://user-images.githubusercontent.com/6216000/141753116-c1fad499-f6cd-4044-bb18-a0ec8fbac490.png">


-------------------


**Post merge**
